spiped/main.c

```diff
@@ -274,50 +284,92 @@ main(int argc, char * argv[])
 			warnp_syslog(1);
 	}
 
+	/* Resolve bind address. */
+	if (opt_b) {
+		char * a = strchr(opt_b, ':');
+		char * b = strrchr(opt_b, ':');
+		char * c = strrchr(opt_b, ']');
+		/* If there is no ':', it's ipv4 address (or Unix domain socket). */
+		if (b == NULL) {
+			if (asprintf(&bind_addr, "%s:0", opt_b) == -1) {
```

Comment:
Do we want to add :0 to a Unix domain socket?  I thought that we'd want to leave a Unix domain socket address unchanged.
(that's what I did in #358, but I might be wrong)